### DISCUSSION TITLE
Migration Widgets

### DISCUSSION THREAD

**Discussion by Lucky-ESA** [moba15/hiob_app#79] [Q&A]
Wie hast du dir die Migration der neuen Widgets vorgestellt? Machst du das über die APP oder müssen die User die Widgets alle neu anlegen? Wollte es gerade per Javascript Adapter machen was leider nicht möglich ist...

**Comment by moba15** [comment]
Sobald ich alle alten widgets durch neue ersetzt habe, werde ich über die App eine Funktion machen, die die alten in neue widgets migriert. Das geht über die App deutlich leichter als über den Adapter. Leider bin ich noch etwas unterwegs die nächsten Tage, aber ich hoffe, dass ich spätestens nächste Woche diesen Punkt erreicht haben werde.
Für den Fall der Fälle wird es aber auch noch einige Zeit möglich sein die alten widgets zu nutzen, sodass bei problemen man immer eine Backup Lösung hätte

**Comment by moba15** [comment]
Sobald ich die oben genannten Punkte noch implementiert habe, muss ich das natürlich alles erstmal gut Testen. Falls aber Interesse besteht könnte ich hier in diesem Thread die PC und apk Version schonmal im vorab hochladen. Somit könntet ihr das schon davor testen.

**Comment by Lucky-ESA** [comment]
Ja, lade mal hoch. Habe meine Table mit einen Adapter umgewandelt. Dabei ist mir ein Fehler (typo) aufgefallen. Pack ich gleich unter Table.

**Comment by moba15** [comment]
Hier wäre die erste vorab version. Es sei jedoch angemerkt, dass es sich hierbei um eine sehr experimentelle version handelt. Also bitte ein backup davor machen.


Man kann jetzt in den Templates settings die alten widgets migrieren. 
[windows-release.zip (GitHub)](https://github.com/user-attachments/files/17096143/windows-release.zip)
[linux-release.zip (GitHub)](https://github.com/user-attachments/files/17096144/linux-release.zip)
[android.zip (workupload)](https://workupload.com/file/a8spgmbMZ6S)

**Comment by Lucky-ESA** [comment]
Habe jetzt die APP Version getestet.
1. Einzeln migrieren -> funktioniert alles einwandfrei
2. Alle selektieren und migrieren -> funktioniert einwandfrei

Folglich, ich konnte keinen Fehler finden. Vielen Dank.
